x-pack/plugins/security/public/plugin.ts

```diff
@@ -28,18 +43,48 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
     this.sessionTimeout = new SessionTimeout(notifications, sessionExpired, http, tenant);
     http.intercept(new SessionTimeoutHttpInterceptor(this.sessionTimeout, anonymousPaths));
 
+    this.license$ = licensing.license$;
+
     return {
       anonymousPaths,
       sessionTimeout: this.sessionTimeout,
     };
   }
 
-  public start() {
+  public start(core: CoreStart) {
     this.sessionTimeout.start();
+
+    const securityLicenseService = new SecurityLicenseService().setup();
+    this.licenseSubscription = this.license$.subscribe(rawLicense => {
+      securityLicenseService.update(rawLicense);
+      const showSecurityLinks = securityLicenseService.license.getFeatures().showLinks;
+
+      const shouldRegisterNavControl =
+        !this.isAnonymousPath(core.http.anonymousPaths) && showSecurityLinks;
+
+      if (shouldRegisterNavControl && !this.navControlRegistered) {
```

Comment:
It's not currently possible to unregister the nav control once it is registered, so this will instead register the nav control the first time it should (based on license and current path). 
Once registered, it will always be visible, even if the license changes in a way where we would want to hide it.

This is functionally equivalent to what existed in the LP, but since the NP is more reactive to configuration changes w/o full page reloads, this might be something we want to address in the future.

Reply:
I need to add a test for this scenario yet

Reply:
Instead of unregistering the nav control, could we always register the nav control and render an empty div when the license changes to no longer enable security?

Reply:
> Instead of unregistering the nav control, could we always register the nav control and render an empty div when the license changes to no longer enable security?

That's not a bad idea. I think I'll keep it as-is for now, so that we don't have a potentially glitchy UI in the case of a networking hiccup or other unexpected issue retrieving license information. I feel like the odds of a connectivity blip are greater than the odds of the license/security settings changing (which would likely require a page reload anyway).